creating new network policy causes kube-router to crash

kubernetes: v1.10.3
kube-router: v0.2.0-beta.7

running kube-router as daemonset, the yaml config is the same as 
kube-router/daemonset/kube-router-firewall-daemonset.yaml

the new network policy's content:
```
apiVersion: networking.k8s.io/v1
kind: NetworkPolicy
metadata:
  namespace: backend
  name: backend
spec:
  podSelector:
    matchLabels:
      app: backend
  policyTypes:
  - Ingress
  ingress:
  - from:
    - podSelector:
        matchLabels:
          app: frontend
    ports:
    - protocol: TCP
      port: 31000
```
when apply this network policy to k8s,kube-router crashed,the kubectl logs {kube-router-pod} shows that:
```
I0704 08:53:28.476403       1 kube-router.go:186] Running /usr/local/bin/kube-router version v0.2.0-beta.7, built on 2018-06-08T09:34:34+0000, go1.8.7
E0704 08:53:29.092510       1 runtime.go:66] Observed a panic: "invalid memory address or nil pointer dereference" (runtime error: invalid memory address or nil pointer dereference)
/home/travis/gopath/src/github.com/cloudnativelabs/kube-router/vendor/k8s.io/apimachinery/pkg/util/runtime/runtime.go:72
/home/travis/gopath/src/github.com/cloudnativelabs/kube-router/vendor/k8s.io/apimachinery/pkg/util/runtime/runtime.go:65
/home/travis/gopath/src/github.com/cloudnativelabs/kube-router/vendor/k8s.io/apimachinery/pkg/util/runtime/runtime.go:51
/home/travis/.gimme/versions/go1.8.7.src/src/runtime/asm_amd64.s:514
/home/travis/.gimme/versions/go1.8.7.src/src/runtime/panic.go:489
/home/travis/.gimme/versions/go1.8.7.src/src/runtime/panic.go:63
/home/travis/.gimme/versions/go1.8.7.src/src/runtime/signal_unix.go:290
/home/travis/gopath/src/github.com/cloudnativelabs/kube-router/pkg/controllers/netpol/network_policy_controller.go:1253
/home/travis/gopath/src/github.com/cloudnativelabs/kube-router/pkg/controllers/netpol/network_policy_controller.go:226
/home/travis/gopath/src/github.com/cloudnativelabs/kube-router/pkg/controllers/netpol/network_policy_controller.go:195
/home/travis/gopath/src/github.com/cloudnativelabs/kube-router/pkg/controllers/netpol/network_policy_controller.go:1456
/home/travis/gopath/src/github.com/cloudnativelabs/kube-router/vendor/k8s.io/client-go/tools/cache/controller.go:195
<autogenerated>:57
/home/travis/gopath/src/github.com/cloudnativelabs/kube-router/vendor/k8s.io/client-go/tools/cache/shared_informer.go:542
/home/travis/gopath/src/github.com/cloudnativelabs/kube-router/vendor/k8s.io/client-go/tools/cache/shared_informer.go:383
/home/travis/gopath/src/github.com/cloudnativelabs/kube-router/vendor/k8s.io/apimachinery/pkg/util/wait/wait.go:71
/home/travis/.gimme/versions/go1.8.7.src/src/runtime/asm_amd64.s:2197
panic: runtime error: invalid memory address or nil pointer dereference [recovered]
	panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x20 pc=0xfda249]

goroutine 103 [running]:
github.com/cloudnativelabs/kube-router/vendor/k8s.io/apimachinery/pkg/util/runtime.HandleCrash(0x0, 0x0, 0x0)
	/home/travis/gopath/src/github.com/cloudnativelabs/kube-router/vendor/k8s.io/apimachinery/pkg/util/runtime/runtime.go:58 +0x126
panic(0x17dc540, 0x27394e0)
	/home/travis/.gimme/versions/go1.8.7.src/src/runtime/panic.go:489 +0x2cf
github.com/cloudnativelabs/kube-router/pkg/controllers/netpol.(*NetworkPolicyController).buildBetaNetworkPoliciesInfo(0xc4206e86e0, 0xc42001ee10, 0x1, 0x1)
	/home/travis/gopath/src/github.com/cloudnativelabs/kube-router/pkg/controllers/netpol/network_policy_controller.go:1253 +0x129
github.com/cloudnativelabs/kube-router/pkg/controllers/netpol.(*NetworkPolicyController).Sync(0xc4206e86e0, 0x0, 0x0)
	/home/travis/gopath/src/github.com/cloudnativelabs/kube-router/pkg/controllers/netpol/network_policy_controller.go:226 +0x659
github.com/cloudnativelabs/kube-router/pkg/controllers/netpol.(*NetworkPolicyController).OnNamespaceUpdate(0xc4206e86e0, 0x1a1b960, 0xc4203bd9c0)
	/home/travis/gopath/src/github.com/cloudnativelabs/kube-router/pkg/controllers/netpol/network_policy_controller.go:195 +0xee
github.com/cloudnativelabs/kube-router/pkg/controllers/netpol.(*NetworkPolicyController).newNamespaceEventHandler.func1(0x1a1b960, 0xc4203bd9c0)
	/home/travis/gopath/src/github.com/cloudnativelabs/kube-router/pkg/controllers/netpol/network_policy_controller.go:1456 +0x3e
github.com/cloudnativelabs/kube-router/vendor/k8s.io/client-go/tools/cache.ResourceEventHandlerFuncs.OnAdd(0xc420799940, 0xc420799950, 0xc420799960, 0x1a1b960, 0xc4203bd9c0)
	/home/travis/gopath/src/github.com/cloudnativelabs/kube-router/vendor/k8s.io/client-go/tools/cache/controller.go:195 +0x49
github.com/cloudnativelabs/kube-router/vendor/k8s.io/client-go/tools/cache.(*ResourceEventHandlerFuncs).OnAdd(0xc4207a12a0, 0x1a1b960, 0xc4203bd9c0)
	<autogenerated>:57 +0x73
github.com/cloudnativelabs/kube-router/vendor/k8s.io/client-go/tools/cache.(*processorListener).run(0xc42077e9b0)
	/home/travis/gopath/src/github.com/cloudnativelabs/kube-router/vendor/k8s.io/client-go/tools/cache/shared_informer.go:542 +0x287
github.com/cloudnativelabs/kube-router/vendor/k8s.io/client-go/tools/cache.(*processorListener).(github.com/cloudnativelabs/kube-router/vendor/k8s.io/client-go/tools/cache.run)-fm()
	/home/travis/gopath/src/github.com/cloudnativelabs/kube-router/vendor/k8s.io/client-go/tools/cache/shared_informer.go:383 +0x2a
github.com/cloudnativelabs/kube-router/vendor/k8s.io/apimachinery/pkg/util/wait.(*Group).Start.func1(0xc4205133a0, 0xc420799ab0)
	/home/travis/gopath/src/github.com/cloudnativelabs/kube-router/vendor/k8s.io/apimachinery/pkg/util/wait/wait.go:71 +0x4f
created by github.com/cloudnativelabs/kube-router/vendor/k8s.io/apimachinery/pkg/util/wait.(*Group).Start
	/home/travis/gopath/src/github.com/cloudnativelabs/kube-router/vendor/k8s.io/apimachinery/pkg/util/wait/wait.go:72 +0x62
```

--- a/pkg/controllers/netpol/network_policy_controller.go
+++ b/pkg/controllers/netpol/network_policy_controller.go
@@ -26,6 +26,7 @@ import (
 	"k8s.io/client-go/kubernetes"
 	listers "k8s.io/client-go/listers/core/v1"
 	"k8s.io/client-go/tools/cache"
+	"regexp"
 )
 
 const (
@@ -1493,6 +1494,8 @@ func NewNetworkPolicyController(clientset kubernetes.Interface,
 
 	npc.v1NetworkPolicy = true
 	v, _ := clientset.Discovery().ServerVersion()
+	valid := regexp.MustCompile("[0-9]")
+	v.Minor = strings.Join(valid.FindAllString(v.Minor, -1), "")
 	minorVer, _ := strconv.Atoi(v.Minor)
 	if v.Major == "1" && minorVer < 7 {
 		npc.v1NetworkPolicy = false

